cookbook/controller/error_pages.rst

```diff
@@ -5,172 +5,154 @@
+The logic followed by the ``ExceptionController`` to pick one of the available
+templates is based on the HTTP status code and request format:
 
-The TwigBundle contains some default templates for error and
-exception pages in its ``Resources/views/Exception`` directory.
+#. Look for a template for the given format and status code (like ``error404.json.twig``
+   or ``error500.xml.twig``);
 
-.. tip::
+#. If the previous template doesn't exist, discard the status code and look for
+   a generic template for the given format (like ``error.json.twig`` or
+   ``error.xml.twig``);
 
-    In a standard Symfony installation, the TwigBundle can be found at
-    ``vendor/symfony/symfony/src/Symfony/Bundle/TwigBundle``. In addition
-    to the standard HTML error page, it also provides a default
-    error page for many of the most common response formats, including
-    JSON (``error.json.twig``), XML (``error.xml.twig``) and even
-    JavaScript (``error.js.twig``), to name a few.
+#. If none of the previous template exist, fall back to the generic HTML template
+   (``error.html.twig``).
 
-Here is how the ``ExceptionController`` will pick one of the
-available templates based on the HTTP status code and request format:
+To override these templates, simply rely on the standard Symfony method for
+:ref:`overriding templates that live inside a bundle <overriding-bundle-templates>`.
+For example, to override the 404 error template for HTML pages, create a new
+``error404.html.twig`` template located at ``app/Resources/TwigBundle/views/Exception/``:
 
-* For *error* pages, it first looks for a template for the given format
-  and status code (like ``error404.json.twig``);
+.. code-block:: html+jinja
 
-* If that does not exist or apply, it looks for a general template for
-  the given format (like ``error.json.twig`` or
-  ``exception.json.twig``);
+    {# app/Resources/TwigBundle/views/Exception/error404.html.twig #}
+    {% extends 'base.html.twig' %}
 
-* Finally, it ignores the format and falls back to the HTML template
-  (like ``error.html.twig`` or ``exception.html.twig``).
+    {% block body %}
+        <h1>Page not found</h1>
 
-.. tip::
+        <p>
+            The requested page couldn't be located. Checkout for any URL
+            misspelling or <a href="{{ path('homepage') }}">return to the homepage</a>.
+        </p>
+    {% endblock %}
 
-    If the exception being handled implements the
-    :class:`Symfony\\Component\\HttpKernel\\Exception\\HttpExceptionInterface`,
-    the ``getStatusCode()`` method will be
-    called to obtain the HTTP status code to use. Otherwise,
-    the status code will be "500".
+Commonly, Symfony applications redefine the ``error404.html.twig`` template, the
+``error500.html.twig`` template for internal server errors and the generic
+``error.html.twig`` template to catch any other error different from 404 and 500.
 
-Overriding or Adding Templates
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
A label for the old headline should be added before the paragraph starting in line 68.